crates/wasmparser/src/validator/component_types.rs

```diff
@@ -1062,7 +1106,13 @@ impl ComponentDefinedType {
                 .any(|case| case.ty.map(|ty| ty.contains_ptr(types)).unwrap_or(false)),
             Self::List(_) => true,
             Self::Tuple(t) => t.types.iter().any(|ty| ty.contains_ptr(types)),
-            Self::Flags(_) | Self::Enum(_) | Self::Own(_) | Self::Borrow(_) => false,
+            Self::Flags(_)
+            | Self::Enum(_)
+            | Self::Own(_)
+            | Self::Borrow(_)
+            | Self::Future(_)
+            | Self::Stream(_)
```

Comment:
I'm a bit surprised by this, how come `contains_ptr` isn't transitive through these types?

Reply:
A `Future` or `Stream` handle is just a u32, like a `Own` resource handle.  The payload type doesn't affect that; the (non)existence of pointers in the payload type only affects how we treat reads to or writes from the stream/future, but has no significance when we're just passing handles around.